laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -261,12 +254,11 @@ public function getChoice()
         $id = $_REQUEST["schedule_id"];
         $choiceTypes = ["First", "Second", "Third"];
         // id is stored as id1_id2_id3, need to split it to get the individual ids
-        $split = explode("_", $id);
+        $schedule_data_ids = explode("_", $id);
         // get current preferences
-        $currentChoices = [];
-        for ($i = 0; $i < 3; $i++) {
-            if ($split[$i] != 0) {
-                $schedule = ScheduleData::where("id", $split[$i])->get();
+        for ($i = 0; $i < count($schedule_data_ids) - 1; $i++) {  //last element of schedule_data_ids is blank
```

Comment:
Can we use `foreach ($schedule_data_ids as $i => $id) {`?

It'd allow us to replace `$schedule_data_ids[$i]` with just `$id`.

Same goes for line [277](https://github.com/DLStahl/CapstoneRemodel/pull/31/files#diff-267411205ec8fcd4ebd1a2f4765f891275d3b4ca661ce213b9346a72e36e1c20R277), [320](https://github.com/DLStahl/CapstoneRemodel/pull/31/files#diff-267411205ec8fcd4ebd1a2f4765f891275d3b4ca661ce213b9346a72e36e1c20R320), and [357](https://github.com/DLStahl/CapstoneRemodel/pull/31/files#diff-267411205ec8fcd4ebd1a2f4765f891275d3b4ca661ce213b9346a72e36e1c20R357) (and any others I may have missed).

Reply:
It would be similar, but I would need to check if $id is an empty string every time so there would be kind of a trade off

Reply:
I don't understand. It seems like you're already checking exactly that on the very next line of code.

Reply:
The if right after logically only checks for 0 in the array (like 6432_5345_0_), and if an empty string is passed into the loop, it creates a 4th element in an array I only want 3 elements in for this specific case. For the loop on line 440, it also would try to delete a 4th option that doesn't exist. Again, I could get around those pretty easily, but I'm not sure the extra logic is worth it.